libselinux/src/selinux_restorecon.c

```diff
@@ -1155,7 +1155,7 @@ void selinux_restorecon_set_sehandle(struct selabel_handle *hndl)
 	fc_sehandle = (struct selabel_handle *) hndl;
 
 	/* Check if digest requested in selabel_open(3), if so use it. */
-	if (selabel_digest(fc_sehandle, &fc_digest, &fc_digest_len,
+	if (fc_sehandle == NULL || selabel_digest(fc_sehandle, &fc_digest, &fc_digest_len,
```

Comment:
IMHO, if !hndl we can just return immediately from selinux_restorecon_set_sehandle() without doing anything.  selabel_no_digest is only ever used if fc_sehandle is non-NULL and fc_sehandle is initially NULL.  It just becomes a no-op.  selinux_restorecon() will then return -1 immediately.

Reply:
ok , 3ks。Is  it means ok if fix the question like this ? 
~~~
void selinux_restorecon_set_sehandle(struct selabel_handle *hndl)
{
	char **specfiles;
	unsigned char *fc_digest;
	size_t num_specfiles, fc_digest_len;
        
        if (!hndl) {
                return;
        }
	fc_sehandle = (struct selabel_handle *) hndl;

	/* Check if digest requested in selabel_open(3), if so use it. */
	if (selabel_digest(fc_sehandle, &fc_digest, &fc_digest_len,
				   &specfiles, &num_specfiles) < 0)
		selabel_no_digest = true;
	else
		selabel_no_digest = false;
}
~~~

Reply:
Probably should return after setting fc_sehandle to hndl just in case anyone uses this to "clear" the fc_sehandle before selabel_close()'ing it.  Also, don't need the { } since it is a single statement body for the if statement.  And I don't see why we have that type cast since it is already that type unless I am missing something.